hdat/hdat_cli.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import traceback
+
+from .resultspec import resolve_resultspecs, print_resultspec
+from .casespec import resolve_casespecs, select_suite
+from .runner import run_cases
+from .util import AbortError
+
+
+def parse_arguments(arguments):
+    parser = argparse.ArgumentParser(prog='hdat')
+    subparsers = parser.add_subparsers(dest='command', metavar='<command>')
+
+    run_help = 'run cases, store results in archive, compare against goldens'
+    run_parser = subparsers.add_parser('run', help=run_help)
+    run_parser.add_argument('casespecs', nargs='*', default=[''], metavar='<case>')
+    run_parser.add_argument('--collect-only', default=False, action='store_true')
+
+    show_help = 'visualize a single result'
+    show_parser = subparsers.add_parser('show', help=show_help)
+    show_result_help = 'result specifier to show'
+    show_parser.add_argument('resultspec', nargs="?", default='', metavar='<result>', help=show_result_help)
+
+    runshow_help = 'run then visualize a single result'
+    runshow_parser = subparsers.add_parser('runshow', help=runshow_help)
+    runshow_parser.add_argument('casespecs', nargs=1, default='', metavar='<result>')
+
+    diff_help = 'compare two results'
+    diff_parser = subparsers.add_parser('diff', help=diff_help)
+    diff_golden_help = 'result to compare to (defaults to current golden for the case)'
```

Comment:
I just noticed this, but in order to do a diff, we need the context, but the context for the current golden set likely won't be available.

I think the easiest solution is to remove the default behavior and make the golden resultspec mandatory.

Reply:
Kind of confused about this one.

Reply:
Sorry, let me expand on my point a bit more.

The CLI currently accepts two arguments.  The first is `goldenspec`, which is a resultspec that should resolve to X results.  The second is `resultspec`, which is also a resultspec that should resolve to Y results.  We expect that X == Y, and should print an error if they do not.  Then, we do something like:

```
for x, y in zip(X, Y):
  suite.diff(x, y)
```

Reply:
New syntax in latest commit is simply 'hdat diff <resultspec> <resultspec>'

Reply:
Nice, looks like `git diff` :)

![screen shot 2018-02-15 at 2 22 45 pm](https://user-images.githubusercontent.com/1648754/36276533-4cdf2d94-125c-11e8-95c3-944b3e0b2d6e.png)